compactor2/src/components/files_split/target_level_non_overlap_split.rs

```diff
@@ -0,0 +1,327 @@
+/// To have better and efficient compaction performance, eligible non-overlapped files
+/// should not be compacted.
+pub struct TargetLevelNonOverlapSplit {}
+
+impl TargetLevelNonOverlapSplit {
+    pub fn new() -> Self {
+        Self {}
+    }
+}
+
+impl Display for TargetLevelNonOverlapSplit {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "Non-overlapping  split for TargetLevel version")
+    }
+}
+
+impl FilesSplit for TargetLevelNonOverlapSplit {
+    /// Return (`[overlapping_files]`, `[non_overlapping_files]`) of the given files
+    /// The non_overlapping_files must in the target_level
+    ///
+    /// Eligible non-overlapping files are files of the target level that do not
+    /// overlap on time range with any lower-level files. All files in target level
+    /// are assumed to not overlap with each other and do not need to check
+    ///
+    /// Example:
+    ///  . Input:
+    ///                          |--L0.1--|             |--L0.2--|
+    ///            |--L1.1--| |--L1.2--|    |--L1.3--|              |--L1.4--|
+    ///
+    ///    (L1.1, L1.3, L1.4) do not overlap with any L0s but only (L1.1, L1.4) are eligible non-overlapping files.
+    ///    (L1.2, L1.3) must be compacted with L0s to produce the right non-overlapping L1s.
+    ///
+    ///  . Output:
+    ///     . overlapping_files: [L0.1, L0.2, L1.2, L1.3]
+    ///     . non_overlapping_files: [L1.1, L1.4]
+    ///
+    /// Algorithm:    
+    ///    The non-overlappings files are files from 2 ends of the target level files that are
+    ///    completely ouside the time range of all lower level files
+    ///
+    ///   L0s                      |--L0.1--|             |--L0.2--|
+    ///   ==> L0s' time range:     |-------L0's time range --------|
+    ///
+    ///   L1s      |--L1.1--| |--L1.2--|    |--L1.3--|              |--L1.4--|
+    ///   ==> Only L1.1 and L1.4 are completely outside the time range of L0s
+    ///
+    fn apply(
+        &self,
+        files: Vec<data_types::ParquetFile>,
+        target_level: CompactionLevel,
+    ) -> (Vec<ParquetFile>, Vec<ParquetFile>) {
+        // Panic if given wrong target level, L0
+        assert_ne!(target_level, CompactionLevel::Initial);
+
+        let mut non_overlapping_files = Vec::with_capacity(files.len());
+        let mut overlapping_files = Vec::with_capacity(files.len());
+
+        // Split files into levels
+        let mut target_level_files = Vec::with_capacity(files.len());
+        let mut prev_level_files = Vec::with_capacity(files.len());
+        let prev_level = target_level.prev();
+        for file in files {
+            if file.compaction_level == target_level {
+                target_level_files.push(file);
+            } else if file.compaction_level == prev_level {
+                prev_level_files.push(file);
+            } else {
+                panic!("Unexpected compaction level: {}", file.compaction_level);
```

Comment:
This means no L2 files, right? Maybe that could be explicitly encoded as a comment
```suggestion
                // L2 files are not supported yet
                panic!("Unexpected compaction level: {}", file.compaction_level);
```